### ISSUE TITLE
Annotate RAG test set with ground truth answer sentences from corpus

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#530]
## Overview
Create ground truth annotations for the RAG test set by identifying the most relevant answer sentences in the corpus for each test question.

## Problem
We have 30 test questions (`data/evaluation/rag_test_set.jsonl`) but no ground truth showing:
- ✅ Which sentences in the corpus actually answer each question
- ✅ Relevance ranking of candidate answer sentences
- ✅ Whether the corpus even contains the answers

Without ground truth, we can't:
- Measure retrieval quality (precision@K, recall@K, MRR, nDCG)
- Distinguish retrieval failures from answer extraction failures
- Train/improve the retrieval component

## Goal
For each of the 30 test questions, identify the **top 50 most relevant sentences** from the corpus and rank them by relevance.

### Phase 1: Automated Candidate Retrieval
Use current retrieval pipeline to find top 100 candidates for each question:

```bash
python scripts/annotate_rag_ground_truth.py \
  --test-set data/evaluation/rag_test_set.jsonl \
  --corpus data/enhanced_corpus/corpus_with_metadata.jsonl \
  --output data/evaluation/rag_ground_truth_candidates.jsonl \
  --top-k 100
```

**Output format:**
```json
{
  "question_id": "factual_001",
  "question": "Kio estas Esperanto?",
  "candidates": [
    {
      "sentence_id": "wikipedia:Esperanto:0:15",
      "text": "Esperanto estas planlingvo...",
      "source": "wikipedia",
      "quality": "BRONZE",
      "retrieval_score": 0.95,
      "retrieval_rank": 1
    },
    ...
  ]
}
```

### Phase 2: Manual Annotation
Review candidates and assign relevance labels:

- **3 - Perfect**: Directly answers the question, could use as-is
- **2 - Good**: Contains answer but needs minor extraction
- **1 - Partial**: Contains relevant info but incomplete
- **0 - Not relevant**: Retrieved but doesn't answer question

**Annotation interface:**
```bash
python scripts/annotate_ground_truth_ui.py \
  --candidates data/evaluation/rag_ground_truth_candidates.jsonl \
  --output data/evaluation/rag_ground_truth_annotated.jsonl
```

### Phase 3: Generate Final Ground Truth
Select top 50 most relevant (relevance >= 1) for each question:

```json
{
  "question_id": "factual_001",
  "question": "Kio estas Esperanto?",
  "ground_truth": [
    {
      "sentence_id": "wikipedia:Esperanto:0:15",
      "text": "Esperanto estas planlingvo kreita de L.L. Zamenhof",
      "relevance": 3,
      "rank": 1,
      "notes": "Perfect definition"
    },
    {
      "sentence_id": "fundamento:intro:0:5",
      "text": "Esperanto estas lingvo internacia...",
      "relevance": 3,
      "rank": 2,
      "notes": "Authoritative definition from Fundamento"
    },
    ...
  ],
  "total_relevant": 47,
  "has_perfect_answer": true,
  "best_source_quality": "GOLD"
}
```

### 1. Retrieval Quality Metrics
```python
# Precision@K: How many of top K retrieved are relevant?
precision_at_10 = relevant_in_top_10 / 10

# Recall@K: What fraction of relevant sentences are in top K?
recall_at_50 = relevant_in_top_50 / total_relevant

# Mean Reciprocal Rank: 1/rank of first relevant sentence
MRR = 1 / rank_of_first_relevant

# nDCG: Normalized discounted cumulative gain
nDCG = compute_ndcg(retrieved_ranks, relevance_scores)
```

### 2. Error Analysis
- **Retrieval failures**: Relevant sentences not retrieved at all
- **Ranking failures**: Relevant sentences retrieved but ranked low
- **Coverage gaps**: No relevant sentences exist in corpus

### 3. Training Data for Retrieval
Use annotated pairs for:
- Fine-tuning retrieval models
- Learning to rank improvements
- Negative sampling (low-scored candidates)

### 4. Corpus Coverage Analysis
```
Category: factual_simple (4 questions)
  ✅ 4/4 have perfect answers (3-rated sentences)
  ✅ Avg 23 relevant sentences per question
  ✅ Best sources: GOLD (fundamento, lingvaj_respondoj)

Category: pronoun_resolution (2 questions)
  ❌ 0/2 have direct answers (expected - needs resolution)
  ⚠️  1/2 have partial answers (relevant context sentences)
```

## Expected Outcomes

For each of 30 questions:
- 📊 **Retrieval metrics**: P@10, R@50, MRR, nDCG
- 📝 **Ground truth**: Top 50 annotated sentences
- ✅ **Coverage**: Whether corpus contains answers
- 🎯 **Best source**: Which corpus source has best answers

**Example metrics table:**
```
Question ID       | P@10 | R@50 | MRR  | Has Answer | Best Source
------------------|------|------|------|------------|-------------
factual_001       | 0.90 | 0.85 | 0.95 | ✅ Yes     | GOLD
grammar_001       | 0.80 | 0.90 | 1.00 | ✅ Yes     | GOLD
pronoun_001       | 0.20 | 0.15 | 0.00 | ❌ No      | N/A
multihop_002      | 0.40 | 0.30 | 0.33 | ⚠️ Partial | SILVER
```

## Implementation Steps

1. **Create annotation script** (`scripts/annotate_rag_ground_truth.py`)
   - Load test set and corpus
   - Run retrieval for each question
   - Output top 100 candidates with scores

2. **Create annotation UI** (`scripts/annotate_ground_truth_ui.py`)
   - Terminal-based review interface
   - Show question + candidate sentences
   - Collect relevance ratings (0-3)
   - Save progress incrementally

3. **Generate final ground truth** (`scripts/finalize_ground_truth.py`)
   - Select top 50 per question
   - Calculate coverage statistics
   - Export in standard format

4. **Update evaluation script** (`scripts/evaluate_rag_test_set.py`)
   - Load ground truth
   - Calculate retrieval metrics
   - Generate metric reports

## Output Files

```
data/evaluation/
├── rag_test_set.jsonl                    # 30 test questions (existing)
├── rag_ground_truth_candidates.jsonl     # Top 100 candidates per question
├── rag_ground_truth_annotated.jsonl      # Manual annotations
├── rag_ground_truth.jsonl                # Final top 50 per question
└── rag_ground_truth_stats.json           # Coverage/quality statistics
```

## Success Criteria

- ✅ All 30 questions have annotated candidates
- ✅ Questions with `expected_performance: "works"` have ≥10 relevant sentences
- ✅ Coverage analysis shows which questions lack answers
- ✅ Can compute retrieval metrics (P@K, R@K, MRR, nDCG)

## Priority

**High** - Blocks proper RAG evaluation. Without ground truth, we can't distinguish good/bad retrieval.

## Related

- Test set created in previous work (30 questions)
- Corpus rebuild in progress (GOLD/SILVER/BRONZE quality)
- Will inform which Stage 3/4 features have highest ROI

## Notes

**Annotation workload:**
- 30 questions × 100 candidates = 3,000 sentence reviews
- At ~5 seconds per review = ~4 hours of annotation work
- Can batch by category (do "works" questions first)
- Can start with top 20 candidates per question (600 reviews = 1 hour)

**Incremental approach:**
1. Annotate "works" questions first (8 questions = 800 reviews)
2. Measure baseline retrieval quality
3. Annotate "partial" questions next
4. Finally annotate "fails" questions (expect low coverage)

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
